comm/src/main/scala/coop/rchain/comm/transport/StreamObservable.scala

```diff
@@ -9,17 +11,24 @@ import monix.reactive.Observable
 import monix.reactive.observers.Subscriber
 import monix.reactive.OverflowStrategy._
 import monix.reactive.subjects.ConcurrentSubject
+import java.nio.file._
 
-case class ToStream(peerNode: PeerNode, blob: Blob)
+case class ToStream(peerNode: PeerNode, path: Path, sender: PeerNode)
 
-class StreamObservable(bufferSize: Int)(implicit log: Log[Task], scheduler: Scheduler)
+class StreamObservable(bufferSize: Int, folder: Path)(implicit log: Log[Task], scheduler: Scheduler)
     extends Observable[ToStream] {
 
   val subject = buffer.LimitedBufferObservable.dropNew[ToStream](bufferSize)
 
   def stream(peers: List[PeerNode], blob: Blob): Task[Unit] = {
     def push(peer: PeerNode): Task[Boolean] =
-      Task.delay(subject.pushNext(ToStream(peer, blob)))
+      blob.packet.store[Task](folder) >>= {
+        case Right(file) => Task.delay(subject.pushNext(ToStream(peer, file, blob.sender)))
+        case Left(UnableToStorePacket(p, er)) =>
+          log.error(s"Could not serialize packet $p. Error message: $er") *> true.pure[Task]
+        case Left(er) =>
+          log.error(s"Could not serialize packet ${blob.packet}. Error: $er") *> true.pure[Task]
```

Comment:
`true` confuses me. could you explain?

Reply:
yeah, I thought `true` might be confusing. 
`push` returns `Boolean`, if it returns `false` then the given packet will be 'retried'. However if we failed to store packet on disk once, we will probably fail again, thus eventually the whole function will loop forever
There is a ticket to introduce sensible retries for stream API (put the message back on the queue, retry few times and discard if impossible - this will be addressed). For now it's safer to log error and drop the message.